pkg/ddl/rollingback.go

```diff
@@ -316,6 +316,46 @@ func rollingbackExchangeTablePartition(jobCtx *jobContext, job *model.Job) (ver
 	return ver, errors.Trace(err)
 }
 
+func rollingbackTruncateTablePartition(jobCtx *jobContext, job *model.Job) (ver int64, err error) {
+	if job.SchemaState == model.StatePublic {
+		job.State = model.JobStateCancelled
+		return ver, dbterror.ErrCancelledDDLJob
+	}
```

Comment:
Could remove those lines. Let the cancel logic all in `convertTruncateTablePartitionJob2RollbackJob` function.